From: Simon McVittie <smcv@debian.org>
Date: Wed, 24 Sep 2025 18:14:48 +0100
Subject: test: Set a much longer timeout for testatomic on GNU/Hurd

The `FIFO test` part of this test-case exercises not only atomic
operations, but also multi-threading more generally, because it has
multiple threads that artificially cause lock contention. libpthread
on the Hurd is mostly implemented in user-space without kernel assistance,
its kernel and user-space are not well-optimized, and Hurd on i386 in
particular only supports one CPU core, so this can be very slow.

Tested-by: Pino Toscano
---
 test/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 074c210..d0be29c 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -377,6 +377,10 @@ add_sdl_test_executable(testdropfile MAIN_CALLBACKS SOURCES testdropfile.c NAME8
 add_sdl_test_executable(testerror NONINTERACTIVE DISABLE_THREADS_ARGS "--no-threads" SOURCES testerror.c NAME83 error)
 add_sdl_test_executable(testsymbols NONINTERACTIVE NOTRACKMEM NONINTERACTIVE_ARGS 0 10 20 40 80 160 320 640 SOURCES testsymbols.c NAME83 symbols)
 
+if(HURD)
+    set_property(TARGET testatomic PROPERTY SDL_NONINTERACTIVE_TIMEOUT 1500)
+endif()
+
 set(build_options_dependent_tests )
 
 add_sdl_test_executable(testevdev BUILD_DEPENDENT NONINTERACTIVE SOURCES testevdev.c NAME83 evdev)
